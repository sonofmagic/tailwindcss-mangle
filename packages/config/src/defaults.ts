import { defaultMangleClassFilter } from '@tailwindcss-mangle/shared'
import type { PatchUserConfig, UserConfig, MangleUserConfig } from './types'

export function getDefaultPatchConfig(): PatchUserConfig {
  return {
    output: {
      filename: '.tw-patch/tw-class-list.json',
      removeUniversalSelector: true,
      loose: true
    },
    tailwindcss: {}
  }
}

export function getDefaultMangleUserConfig(): MangleUserConfig {
  return {
    mangleClassFilter: defaultMangleClassFilter,
    include: ['**/*.{js,jsx,ts,tsx,svelte,vue}'],
    exclude: ['node_modules/**/*', '**/*.{css,scss,less,sass,postcss,html,htm}'],
    disabled: process.env.NODE_ENV === 'development',
    classListPath: '.tw-patch/tw-class-list.json',
    classMapOutput: {
      enable: false,
      filename: '.tw-patch/tw-map-list.json',
      loose: true
    }
  }
}

export function getDefaultUserConfig(): UserConfig {
  return {
    patch: getDefaultPatchConfig(),
    mangle: getDefaultMangleUserConfig()
  }
}
