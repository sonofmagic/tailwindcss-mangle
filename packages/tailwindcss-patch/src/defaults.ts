import type { PatchOptions, UserConfig, DeepRequired } from './types'

export function getDefaultPatchOptions(): DeepRequired<PatchOptions> {
  return {
    overwrite: true
  }
}

export function getDefaultUserConfig(): UserConfig {
  return {
    output: {
      filename: '.tw-patch/tw-class-list.json',
      removeUniversalSelector: true,
      loose: true
    },
    tailwindcss: {
      cwd: process.cwd()
    }
  }
}
