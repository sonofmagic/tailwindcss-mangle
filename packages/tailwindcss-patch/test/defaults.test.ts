import { getDefaultPatchOptions, getDefaultUserConfig } from '@/defaults'

describe('defaults', () => {
  it('getDefaultPatchOptions', () => {
    expect(getDefaultPatchOptions()).toMatchSnapshot()
  })

  it('getDefaultUserConfig', () => {
    expect(getDefaultUserConfig().output).toMatchSnapshot()
  })
})
