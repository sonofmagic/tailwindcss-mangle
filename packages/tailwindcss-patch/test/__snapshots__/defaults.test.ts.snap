// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaults > getDefaultPatchOptions 1`] = `
{
  "overwrite": true,
}
`;

exports[`defaults > getDefaultUserConfig 1`] = `
{
  "filename": ".tw-patch/tw-class-list.json",
  "loose": true,
  "removeUniversalSelector": true,
}
`;
